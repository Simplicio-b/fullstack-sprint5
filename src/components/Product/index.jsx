import { Link } from 'react-router-dom';

import { ProductsCard, Card, ProductImg } from './styles';

function Product({ image, name, price, sku }) {
	return (
		<ProductsCard>
			<Link to={`product-detail/${sku}`}>
				<Card>
					<ProductImg className="card__img" src={image} alt="" />
					<p className="card__description">{name}</p>
					<p className="card__price" data-testid="product-price-testid">
						R$ {price}
					</p>
				</Card>
			</Link>
		</ProductsCard>
	);
}

export default Product;
