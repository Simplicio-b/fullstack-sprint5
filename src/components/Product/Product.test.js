import { render, screen } from '@testing-library/react';
import Product from './index';
import { BrowserRouter, Route } from 'react-router-dom';

describe('Testando component Product', () => {
	it('Teste de snapshot', () => {
		render(
			<BrowserRouter>
				<Route>
					<Product
						image="assets/404.svg"
						name="Cuzcus"
						price="199,50"
						sku={1}
					/>
				</Route>
			</BrowserRouter>
		);

		expect(screen.getByText('Cuzcus')).toMatchSnapshot();
	});

	it('Teste de renderizacao', () => {
		render(
			<BrowserRouter>
				<Route>
					<Product
						image="assets/c12.webp"
						name="Camiseta Infantil"
						price="39,90"
						sku={12}
					/>
				</Route>
			</BrowserRouter>
		);

		expect(screen.getByText('Camiseta Infantil')).toBeInTheDocument();
		expect(screen.getByTestId('product-price-testid').textContent).toBe(
			'R$ 39,90'
		);
	});
});
