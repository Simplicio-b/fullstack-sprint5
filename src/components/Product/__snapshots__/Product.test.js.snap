// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando component Product Teste de snapshot 1`] = `
<p
  class="card__description"
>
  Cuzcus
</p>
`;
